linker/linker_phdr.cpp

```diff
@@ -711,7 +793,8 @@ static int _phdr_table_set_load_prot(const ElfW(Phdr)* phdr_table, size_t phdr_c
   const ElfW(Phdr)* phdr_limit = phdr + phdr_count;
 
   for (; phdr < phdr_limit; phdr++) {
-    if (phdr->p_type != PT_LOAD || (phdr->p_flags & PF_W) != 0) {
+    if (phdr->p_type != PT_LOAD || (phdr->p_flags & PF_W) != 0 ||
+        (phdr->p_flags & PF_RAND_ADDR) != 0) {
```

Comment:
`(phdr->p_flags & (PF_W | PF_RAND_ADDR))` would be shorter, but I don't know if it's cleaner?!